### PR TITLE
init

### PR DESCRIPTION
**Pull request by JanTvrdik**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,11 @@
+* text=auto
+
+/.github export-ignore
+/tests export-ignore
```
Comment (janedbal) [inline .gitattributes:4]: `/cache export-ignore`

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "scripts": {
+        "check": [
+            "@check:composer",
+            "@check:ec",
+            "@check:cs",
+            "@check:types",
+            "@check:tests",
+            "@check:dependencies"
+        ],
+        "check:composer": "composer normalize --dry-run --no-check-lock --no-update-lock",
```
Comment (janedbal) [inline composer.json:53]: add also `composer validate --strict`

File: phpstan.neon.dist [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    - ./vendor/shipmonk/phpstan-rules/rules.neon
+
+parameters:
+    paths:
+        - src
+        - tests
+    checkMissingCallableSignature: true
+    checkUninitializedProperties: true
+    checkTooWideReturnTypesInProtectedAndPublicMethods: true
+    checkBenevolentUnionTypes: true
```
Comment (janedbal) [inline phpstan.neon.dist:16]: Consider adding [even more strictness](https://github.com/shipmonk-rnd/phpstan-rules?tab=readme-ov-file#native-phpstan-extra-strictness)

Comment (janedbal) [reply phpstan.neon.dist:16]: Hmm, but with level8, it might be hard.

File: phpstan.neon.dist [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        check:
+            missingCheckedExceptionInThrows: true
+            tooWideThrowType: true
+        implicitThrows: false
+        checkedExceptionClasses:
+            - ShipMonk\DoctrineEntityPreloader\Exception\RuntimeException
+
+    shipmonkRules:
+        allowNamedArgumentOnlyInAttributes:
+            enabled: false
```
Comment (janedbal) [inline phpstan.neon.dist:28]: This was already dropped, use v3

File: src/EntityPreloader.php [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+use function get_parent_class;
+use function is_a;
+
+/**
+ * @template E of object
+ */
+class EntityPreloader
+{
+
+    private const BATCH_SIZE_TO_AVOID_FULL_INDEX_SCAN = 1_000;
```
Comment (janedbal) [inline src/EntityPreloader.php:24]: I believe this constant is wrongly named. There is no full index scan at risk imo. So maybe just BATCH_SIZE?

**Review by janedbal — CHANGES_REQUESTED** [review]

File: phpstan.neon.dist (hunk 3 above)
Comment (JanTvrdik) [reply phpstan.neon.dist:16]: maybe in the future with level 9

**Comment by JanTvrdik** [comment]
PR was updated. See [changes](https://github.com/shipmonk-rnd/doctrine-entity-preloader/compare/1aafbdd6062bfdad046d4217ef3756574c183502..5c0eb05)

**Review by janedbal — APPROVED** [review]

**Pull request merged**